src/core/memory.d

```diff
@@ -809,3 +809,41 @@ struct GC
         gc_runFinalizers( segment );
     }
 }
+
+/// Pure variant of `malloc` that doesn't affect global variable `errno`.
```

Comment:
Please replace this with the following:

```
/**
 * Pure variants of C's memory allocation functions. Purity is achieved via
 * resetting the `errno` to it's value prior to being called, removing the
 * function's global state mutation.
 *
 * See_Also:
 *     $(LINK2 https://dlang.org/spec/function.html#pure-functions, D's rules for purity),
 *     which allow for memory allocation under specific circumstances.
 */
```

IMO it addresses the intuitiveness of a pure memory allocation function for the reader.